airflow/providers/amazon/aws/config_templates/config.yml

```diff
@@ -0,0 +1,131 @@
+      type: boolean
+      example: "True"
+      default: "False"
+    cluster:
+      description: |
+        Name of the Amazon ECS Cluster. Required.
+      version_added: "2.8"
+      type: string
+      example: "ecs_executor_cluster"
+      default: ~
+    container_name:
+      description: |
+        Name of the container that will be used to execute Airflow tasks via the ECS executor.
+        The container should be specified in the ECS Task Definition and will receive an airflow
+        CLI command as an additional parameter to its entrypoint. For more info see url to Boto3
+        docs above. Required.
+      version_added: "2.8"
+      type: string
+      example: "ecs_executor_container"
+      default: ~
+    launch_type:
+      description: |
+        Launch type can either be 'FARGATE' OR 'EC2'. For more info see url to
+        Boto3 docs above.
+
+        If the launch type is EC2, the executor will attempt to place tasks on
+        empty EC2 instances. If there are no EC2 instances available, no task
+        is placed and this function will be called again in the next heart-beat.
+
+        If the launch type is FARGATE, this will run the tasks on new AWS Fargate
+        instances.
+      version_added: "2.8"
+      type: string
+      example: "FARGATE"
+      default: "FARGATE"
+    platform_version:
+      description: |
+        The platform version the task uses. A platform version is only specified
+        for tasks hosted on Fargate. If one isn't specified, the LATEST platform
+        version is used.
+      version_added: "2.8"
+      type: string
+      example: "1.4.0"
+      default: "LATEST"
+    security_groups:
+      description: |
+        The comma-seperated IDs of the security groups associated with the task. If you
+        don't specify a security group, the default security group for the VPC is used.
+        There's a limit of 5 security groups. For more info see url to Boto3 docs above.
+      version_added: "2.8"
+      type: string
+      example: "sg-XXXX,sg-YYYY"
+      default: ~
+    subnets:
+      description: |
+        The comma-separated IDs of the subnets associated with the task or service.
+        There's a limit of 16 subnets. For more info see url to Boto3 docs above.
+      version_added: "2.8"
+      type: string
+      example: "subnet-XXXXXXXX,subnet-YYYYYYYY"
+      default: ~
+    task_definition:
+      description: |
+        The family and revision (family:revision) or full ARN of the task definition
+        to run. If a revision isn't specified, the latest ACTIVE revision is used.
+        For more info see url to Boto3 docs above.
+      version_added: "2.8"
+      type: string
+      example: executor_task_definition:LATEST
+      default: ~
+    max_run_task_attempts:
+      description: |
+        The maximum number of times the Ecs Executor should attempt to run a task.
+      version_added: "2.8"
+      type: int
+      example: "3"
+      default: "3"
```

Comment:
This feels like a dangerous default (and maybe even feature). Don't we want Airflows retries to control this instead?

Reply:
We debated about this one a lot.

This mechanism controls retrying the _attempt to run the task_, e.g. interacting with the ECS API, containers starting up correctly, etc. All the things before the Airflow task code even begins to run, so there should be no idempotentcy issues. Once the Airflow task is up and successfully running, then it's Airflow retries that take over.

Reply:
Gotcha. If you're confident this just covers retrying up till any Airflow code is run, I think that's fine. It'd be nice to clarify in the description that's how it behaves though.

Reply:
We'll see how it behaves once it's merged and we have some alpha users. We've also updated the description to clarify which retries exactly this configures.